### ISSUE TITLE
P1 Smart Meter cannot deal with info added to telegram in latest Belgian meter update

### ISSUE THREAD

**Issue by JanCeuleers** [domoticz/domoticz#5909]
As discussed on the forum [1], my P1 smart meter stopped being read by Domoticz following a firmware update of the meter that took effect 2 days ago at midnight. The log merely showed once per minute that Domoticz had not received data from the meter and that the module was being restarted. This was with only Error reporting enabled; once also Info and Status were enabled the following was logged:

2023-12-02 10:32:38.506 Error: Fluvius hardware (10) nothing received for more than 1 Minute!....
2023-12-02 10:32:39.506 Status: Fluvius: TCP/IP Worker stopped...
2023-12-02 10:32:39.507 Status: Fluvius: attempt connect to meterpi.xperim.be:12345
2023-12-02 10:32:39.509 Status: Fluvius: connected to: meterpi.xperim.be:12345
2023-12-02 10:32:39.866 Status: Fluvius: Meter reports as ESMR 5.0.2
2023-12-02 10:32:39.962 Status: Fluvius: Found gas meter on M-Bus channel 1
2023-12-02 10:32:39.962 Status: Fluvius: Found 'Water' meter on M-Bus channel
2023-12-02 10:32:39.506 Error: Restarting: Fluvius

This demonstrates that the meter info does in fact get through to Domoticz, since it sees the meter and the subtending gas and water meters as well, yet it doesn't accept the data being sent.

Here is a telegram captured from the meter (after the firmware update):

0-0:96.1.4(50217)
0-0:96.1.1(3153414733313030343733313736)
0-0:1.0.0(231201200610W)
1-0:1.8.1(002223.828*kWh)
1-0:1.8.2(005377.254*kWh)
1-0:2.8.1(001497.794*kWh)
1-0:2.8.2(000647.242*kWh)
0-0:96.14.0(0001)
1-0:1.4.0(00.261*kW)
1-0:1.6.0(231201063000W)(06.334*kW)
0-0:98.1.0(12)(1-0:1.6.0)(1-0:1.6.0)(230101000000W)(221220043000W)(03.539*kW)(230)
1-0:1.7.0(00.650*kW)
1-0:2.7.0(00.000*kW)
1-0:21.7.0(00.098*kW)
1-0:41.7.0(00.244*kW)
1-0:61.7.0(00.307*kW)
1-0:22.7.0(00.000*kW)
1-0:42.7.0(00.000*kW)
1-0:62.7.0(00.000*kW)
1-0:32.7.0(224.5*V)
1-0:52.7.0(226.1*V)
1-0:72.7.0(225.5*V)
1-0:31.7.0(001.30*A)
1-0:51.7.0(001.96*A)
1-0:71.7.0(001.71*A)
0-0:96.3.10(1)
0-0:17.0.0(999.9*kW)
1-0:31.4.0(999*A)
0-0:96.13.0()
0-1:24.1.0(003)
0-1:96.1.1(37464C4F32313231313833353734)
0-1:24.4.0(1)
0-1:24.2.3(231201200459W)(03304.972*m3)
0-2:24.1.0(007)
0-2:96.1.1(3853455430303030393630383636)
0-2:24.2.1(231201200527W)(00009.138*m3)
!9382

Some collaborative detective work on the forum reveals that the code does not expect the 1.4.0, 1.6.0 and 98.1.0 lines.

So I inserted a "grep -v" in between the meter and Domoticz that filters those lines out, and it works again, confirming the above. Details in the forum thread.

[1]: https://www.domoticz.com/forum/viewtopic.php?p=309462#p309462

Version: 2023.2 (build 15457)
Build Hash: 25624ca14
Compile Date: 2023-07-22 10:39:06
dzVents Version: 3.1.8
Python Version: 3.8.10 (default, Nov 22 2023, 10:22:35) [GCC 9.4.0]

Platform: Ubuntu 20.04

**Comment by CvvtLo** [comment]
I experience the same problem, however i am using a P1 USB cable to capture the smart meter  data.

The error message I get is : "Dismiss incoming - message oversized" ( line 1284 of the P1MeterBase.cpp code file )

![image](https://github.com/domoticz/domoticz/assets/46109429/427ee90b-dd8d-4c99-a0b8-62ce0d96fa72)


Here a full dump of a P1 message from my smart meter: 

/FLU5\253769484_A

0-0:96.1.4(50217)
0-0:96.1.1(315341473331
0-0:1.0.0(231202162612)
1-0:1.8.1(004729.292*kWh)
1-0:1.8.2(007541.700*kWh)
1-0:2.8.1(010741.503*kWh)
1-0:2.8.2(004282.459*kWh)
0-0:96.14.0(0002)
1-0:1.4.0(00.374*kW)
1-0:1.6.0(231201110000W)(01.701*kW)
0-0:98.1.0(12)(1-0:1.6.0)(1-0:1.6.0)(230101000000W)(221224134500W)(05.594*kW)(230201000000W)(230101140000W)(04.642*kW)(230301000000W)(230204191500W)(03.298*kW)(230401000000S)(230310184500W)(02.918*kW)(230501000000S)(230407181500S)(02.751*kW)(230601000000S)(230511194500S)(02.792*kW)(230701000000S)(230619190000S)(02.889*kW)(230801000000S)(230701183000S)(03.053*kW)(230901000000S)(230817200000S)(03.114*kW)(231001000000S)(230912183000S)(03.457*kW)(231101000000W)(231025181500S)(03.283*kW)(231201000000W)(231115180000W)(04.374*kW)
1-0:1.7.0(00.528*kW)
1-0:2.7.0(00.000*kW)
1-0:21.7.0(00.020*kW)
1-0:41.7.0(00.000*kW)
1-0:61.7.0(00.508*kW)
1-0:22.7.0(00.000*kW)
1-0:42.7.0(00.000*kW)
1-0:62.7.0(00.000*kW)
1-0:32.7.0(237.1*V)
1-0:52.7.0(234.3*V)
1-0:72.7.0(233.0*V)
1-0:31.7.0(000.33*A)
1-0:51.7.0(000.71*A)
1-0:71.7.0(002.54*A)
0-0:96.3.10(1)
0-0:17.0.0(999.9*kW)
1-0:31.4.0(999*A)
0-0:96.13.0()
0-1:24.1.0(003)
0-1:96.1.1(37464C4F32313139313231303233)
0-1:24.4.0(1)
0-1:24.2.3(632525252525W)(00000.000*m3)
0-2:24.1.0(003)
0-2:96.1.1(37464C4F32313230303134303733)
0-2:24.4.0(1)
0-2:24.2.3(231202162504W)(07070.831*m3)
!6528


Fact is that the 0-0:98  line is much longer than 400 characters , which is probably the cause of the problem.

**Comment by JanCeuleers** [comment]
I did not get that message, but I can confirm that it is enough to filter the 98.1.0 line for Domoticz to accept the telegram. I will add a suggestion to the forum post about this, and also a suggested work-around for USB users.

**Comment by waltervl** [comment]
I see that @gizmocuz increased the buffer for P1 meter recently so perhaps this will solve the issue? It should be in latest beta. 
https://github.com/domoticz/domoticz/commit/7d7e2c0633d6bb7726ef4d5193bf8f7702a78fcd

**Comment by JanCeuleers** [comment]
It seems that there are at least two problems.

Upon closer inspection the 98.1.0 line that my meter emits is far less long than 400 characters, which explains why I wasn't seeing the "message oversized" error. But my meter readings were still being silently rejected by Domoticz.

So upping the maximum length of any single line to 600 characters, and also increasing the overall buffer size from 1400 to 2000 characters may address part of the problem but not all of it.

Could I suggest that the developers test with the telegram posted above, and also the one from the forum?

**Comment by gizmocuz** [comment]
I did test with the telegram above.
We are not using 98.1.0 at the moment, no idea what it is, but maybe it could be useful someday.
And limiting the line width would truncate this line
Did you test with the latest beta?

**Comment by JanCeuleers** [comment]
I have not tested with the latest beta. I am very reluctant to install the beta because this is my production system and I have been very severely bitten in the past by upgrading Domoticz only to find that the newer version had upped its dependencies such that I also had to upgrade my distro in order to return Domoticz to service. I can't risk that again.

If you can point me to instructions for testing the latest beta in a way that is going to work on my distro (Ubuntu 20.04) and that I am guaranteed to be able to roll back, then that would be welcome.

Meanwhile I can reinforce my previous point: the telegram produced by my meter contains a 98.1.0 line that is shorter than 400 characters, and the overall size of the telegram is also only around 925 bytes. Here is an example:

0-0:96.1.4(50217)
0-0:96.1.1(3153414733313030343733313736)
0-0:1.0.0(231201200610W)
1-0:1.8.1(002223.828*kWh)
1-0:1.8.2(005377.254*kWh)
1-0:2.8.1(001497.794*kWh)
1-0:2.8.2(000647.242*kWh)
0-0:96.14.0(0001)
1-0:1.4.0(00.261*kW)
1-0:1.6.0(231201063000W)(06.334*kW)
0-0:98.1.0(12)(1-0:1.6.0)(1-0:1.6.0)(230101000000W)(221220043000W)(03.539*kW)(230)
1-0:1.7.0(00.650*kW)
1-0:2.7.0(00.000*kW)
1-0:21.7.0(00.098*kW)
1-0:41.7.0(00.244*kW)
1-0:61.7.0(00.307*kW)
1-0:22.7.0(00.000*kW)
1-0:42.7.0(00.000*kW)
1-0:62.7.0(00.000*kW)
1-0:32.7.0(224.5*V)
1-0:52.7.0(226.1*V)
1-0:72.7.0(225.5*V)
1-0:31.7.0(001.30*A)
1-0:51.7.0(001.96*A)
1-0:71.7.0(001.71*A)
0-0:96.3.10(1)
0-0:17.0.0(999.9*kW)
1-0:31.4.0(999*A)
0-0:96.13.0()
0-1:24.1.0(003)
0-1:96.1.1(37464C4F32313231313833353734)
0-1:24.4.0(1)
0-1:24.2.3(231201200459W)(03304.972*m3)
0-2:24.1.0(007)
0-2:96.1.1(3853455430303030393630383636)
0-2:24.2.1(231201200527W)(00009.138*m3)
!9382

**Comment by gizmocuz** [comment]
I still don't understand what you are trying to proof? You also miss the first/start line

I am not sure if you are able to test the beta, as this is a pretty straight forward procedure...
Stopping domoticz, make a full backup of the folder, issue ./updatebeta in the domoticz folder
If it is not working correctly (which I highly doubt!) you can stop it and replace it with your backup

As alternative you can install Domoticz via docker, a link for this is on the download page

**Comment by JanCeuleers** [comment]
Thank you for the instructions.

I have tested the beta, and it does indeed solve the problem for me.

As someone else pointed out on the forum, upping the maximum line length to 600 chars may only kick the can down the road, unless the standard guarantees that the line length will remain below that value.

(This is now a moot point, but what I was trying to say is that the telegram produced by my meter should have already been accepted because its total length is below 1400 and each line is also shorter than 400 characters. So I don't understand how this change makes it work; perhaps there are other changes in the beta relative to the latest stable version that can explain this?)

(EDIT: I have looked at my meter's telegram again, and now the 98.1.0 line is indeed very long. Case closed, thanks).
